micro-rdk/src/esp32/entry.rs

```diff
@@ -149,6 +150,9 @@ pub async fn serve_web_inner(
             max_webrtc_connection,
         )
         .with_webrtc(webrtc)
+        .with_periodic_app_client_task(Box::new(RestartMonitor::new(|| unsafe {
```

Comment:
The payoff for this approach is that we can just dependency inject the esp2 specific restart in the esp32 `entry.rs`, and a different one in the native `entry.rs`.